src/software/ai/hl/stp/stp.cpp

```diff
@@ -182,15 +206,21 @@ bool STP::overrideAIPlayIfApplicable()
 
 
 std::map<std::shared_ptr<const Tactic>, Robot> STP::assignRobotsToTactics(
-    ConstPriorityTacticVector tactics, const World& world)
+    ConstPriorityTacticVector tactics, const World& world,
+    bool automatically_assign_goalie)
 {
-    std::map<std::shared_ptr<const Tactic>, Robot> robot_tactic_assignment;
+    robot_tactic_assignment.clear();
 
     std::optional<Robot> goalie_robot = world.friendlyTeam().goalie();
     std::vector<Robot> robots         = world.friendlyTeam().getAllRobots();
 
-    auto is_goalie_tactic = [](auto tactic) { return tactic->isGoalieTactic(); };
-    bool goalie_assigned  = false;
+    if (goalie_robot && automatically_assign_goalie)
+    {
```

Comment:
In what game scenarios would `automatically_assign_goalie` be false?

Reply:
example play and penalty kick